plugin_runner/plugin_runner.py

```diff
@@ -182,6 +186,28 @@ async def ReloadPlugins(
             yield ReloadPluginsResponse(success=True)
 
 
+def apply_effects_to_context(effects: list[Effect], event: Event) -> Event:
+    """Applies AUTOCOMPLETE_SEARCH_RESULTS effects to the event context.
+    If we are dealing with a search event, we need to update the context with the search results.
+    """
+    # Skip if the event is not a search event
+    if not EventType.Name(event.type).endswith("__PRE_SEARCH") and not EventType.Name(
+        event.type
+    ).endswith("__POST_SEARCH"):
+        return event
+
+    context = json.loads(event.context)
+    for effect in effects:
+        if effect.type == EffectType.AUTOCOMPLETE_SEARCH_RESULTS:
+            context["results"] = json.loads(effect.payload)
+            # Stop processing effects if we've found a AUTOCOMPLETE_SEARCH_RESULTS
```

Comment:
In retro we discussed how users might return incompatible effects from a plugin. We can't predict all the ways that this might happen, but this is a pretty obvious one.

It's always easier to enable something later than it is to disable it later, so disabling this right now would be sensible and also the fastest way to resolve this right now.

Reply:
@csande At what point do you think we should validate this? Right after a plugin returns its effects? If a plugin produces incompatible effects, should we simply log a warning and discard the returned effects, or do you think we should take a different approach?

Reply:
ended up implementing `validate_effects` that runs after every protocol execution